src/mbgl/layout/symbol_layout.cpp

```diff
@@ -738,10 +738,12 @@ std::vector<float> SymbolLayout::calculateTileDistances(const GeometryCoordinate
                 sumForwardLength += util::dist<float>(line[i + 1], line[i]);
             }
         }
-        for (std::size_t i = ++segment; i-- > 0;) {
+        for (std::size_t i = segment;; --i) {
```

Comment:
for loop without condition is odd. Could it be added, even if not needed? (so that the break can also go)

Reply:
`while(--i )`